examples/with-drizzle/src/app/page.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+
+import { useState } from "react";
+
+import { UploadButton, UploadDropzone } from "~/utils/uploadthing";
+
+export default function Home() {
+  const [files, setFiles] = useState([]);
+
+  const fetchFiles = () => {
+    fetch("/api/files")
+      .then((res) => res.json())
+      .then((res) => setFiles(res.files));
+  };
+
+  return (
+    <main>
+      <UploadButton
+        /**
+         * @see https://docs.uploadthing.com/api-reference/react#uploadbutton
+         */
+        endpoint="videoAndImage"
+        onClientUploadComplete={(res) => {
+          console.log(`onClientUploadComplete`, res);
+          alert("Upload Completed");
+
+          setTimeout(() => fetchFiles(), 300);
```

Comment:
this pattern looks a bit weird, might wanna polish it a bit. should probably also fetch the files in a server component?

Reply:
The goal was to receive up-to-date list of files. I added small timeout to fetching in case there onClientUploadComplete is called too soon. The fix for dev hook was not merged yet, so the order of actions is unreliable

I was thinking about RSC for a bit, but decided to go with the simplest way. This example is about integration between DB and UT uploads

Reply:
So I don't think we need a fancy ways of retrieving data on client

Reply:
just using a timeout is inconsistent at best, we should probably show an example of polling, either with our logic, or by leveraging something like SWR

Reply:
In this case, SWR should suffice. It has `refreshInterval` which can be set to like 250ms. Link https://swr.vercel.app/docs/revalidation#revalidate-on-interval

Reply:
Yup!

Reply:
i think for the appdir example we should do the fetching in server components (as it is now) - you can use swr in the pages/ example if the plan is to add that too

Reply:
Now that I thought a bit more about it, I agree we should try to encourage server components in any appdir examples, since they will provide the best end user experience.